### PR TITLE
Sync more develop changes to hdf5_1_14

### PR DESCRIPTION
**Pull request by lrknox**

### CODE REVIEW

File: .github/workflows/daily-build.yml [hunk 1]
```diff
@@ -12,6 +12,26 @@ permissions:
...
+    outputs:
+      hdf5-name: ${{ steps.gethdf5base.outputs.HDF5_NAME_BASE }}
+
+    steps:
+    - uses: actions/checkout@v4.1.1
+
+    - name: Get hdf5 release base name
+      uses: dsaltares/fetch-gh-release-asset@master
+      with:
+        version: 'tags/snapshot'
```
Comment (byrnHDF) [inline .github/workflows/daily-build.yml]: snapshot-1.14

File: .github/workflows/daily-build.yml [hunk 2]
```diff
@@ -50,3 +70,14 @@ jobs:
...
     if: ${{ needs.call-workflow-tarball.outputs.has_changes == 'true' }}
 
+  call-workflow-remove:
+    needs: [get-old-names, call-workflow-tarball, call-workflow-ctest, call-workflow-abi, call-workflow-release]
+    permissions:
+      contents: write # In order to allow file deletion
+    uses: ./.github/workflows/remove-files.yml
+    with:
+      file_base: ${{ needs.get-old-names.outputs.hdf5-name }}
+      use_tag: snapshot
```
Comment (byrnHDF) [inline .github/workflows/daily-build.yml]: snapshot-1.14

File: .github/workflows/daily-build.yml [hunk 3]
```diff
@@ -12,6 +12,26 @@ permissions:
...
+
+    - name: Read base-name file
+      id: gethdf5base
+      run: echo "HDF5_NAME_BASE=$(cat last-file.txt)" >> $GITHUB_OUTPUT
+
+    - run: echo "hdf5 base name is ${{ steps.gethdf5base.outputs.HDF5_NAME_BASE }}."
+
   call-workflow-tarball:
     uses: ./.github/workflows/tarball.yml
     with:
```
Comment (byrnHDF) [inline .github/workflows/daily-build.yml:37]: line 26-46 insert line before "#usetag" preset_name: ci-StdShar

Comment (byrnHDF) [reply .github/workflows/daily-build.yml:37]: cmake-ctest.yml needs fixed - only half have the correct variable.

File: .github/workflows/daily-build.yml [hunk 4]
```diff
@@ -12,9 +12,30 @@ permissions:
...
+      preset_name: ci-StdShar
```
Comment (byrnHDF) [inline .github/workflows/daily-build.yml]: this is one too many

File: .github/workflows/daily-build.yml [hunk 5]
```diff
@@ -23,6 +44,7 @@ jobs:
     uses: ./.github/workflows/cmake-ctest.yml
     with:
       file_base: ${{ needs.call-workflow-tarball.outputs.file_base }}
+      preset_name: ci-StdShar
```
Comment (byrnHDF) [inline .github/workflows/daily-build.yml:46]: This is correct

File: .github/workflows/daily-build.yml (hunk 4 above)
Comment (lrknox) [inline .github/workflows/daily-build.yml]:
```suggestion
```

**Review by byrnHDF — APPROVED** [review]

**Pull request merged**